lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1114,45 +1145,79 @@ def _get_app_or_unit(self) -> Union[Application, Unit]:
     @property
     def private_key(self) -> Optional[PrivateKey]:
         """Return the private key."""
+        if self._private_key:
+            return self._private_key
         if not self._private_key_generated():
             return None
         secret = self.charm.model.get_secret(label=self._get_private_key_secret_label())
         private_key = secret.get_content(refresh=True)["private-key"]
         return PrivateKey.from_string(private_key)
 
-    def _generate_private_key(self) -> None:
+    def _ensure_private_key(self) -> None:
+        """Make sure there is a private key to be used.
+
+        It will make sure there is a private key provided by the charm otherwise it will
+        generate a new one.
```

Comment:
Again, this "provided by the charm" wording is confusing I find - the lib is part of the charm.

Reply:
Reworded.